### PR TITLE
#3068 Experience names with emojis are displayed incompletely

### PR DESCRIPTION
**Pull request by LLGuru**

### CODE REVIEW

**Review by marchcat — APPROVED** [review]

**Review by akleshchev — APPROVED** [review]

**Comment by akleshchev** [comment]
How is this related to emojis?

File: indra/llrender/llfontgl.cpp [hunk 1]
```diff
@@ -336,16 +341,7 @@ S32 LLFontGL::render(const LLWString &wstr, S32 begin_offset, F32 x, F32 y, cons
...
-        {
-            gGL.begin(LLRender::TRIANGLES);
-            {
-                gGL.vertexBatchPreTransformed(vertices, uvs, colors, glyph_count * 6);
-            }
-            gGL.end();
-
-            glyph_count = 0;
-        }
+        draw_queued_glyphs();
```
[the remark this answers is not in the document]
Comment (LLGuru) [reply indra/llrender/llfontgl.cpp:344]: > Wouldn't this result in the GLYPH_BATCH_SIZE being ignored?

This is the core of the current bug fix

[the remark this answers is not in the document]
Comment (akleshchev) [reply indra/llrender/llfontgl.cpp:344]: I missed that you removed (glyph_count >= GLYPH_BATCH_SIZE) check...
The purpose of this check is to not overfill buffer. If you let viewer run a string that is too large without limiting batch size,  vertexBatchPreTransformed will detect an overflow and will not draw.

[the remark this answers is not in the document]
Comment (akleshchev) [reply indra/llrender/llfontgl.cpp:344]: This commit is likely to cause bugs with long strings.

**Pull request merged**